### PR TITLE
Added unittests for the OpenAIStreamService

### PR DESCRIPTION
**Pull request by ahiipsa**

### CODE REVIEW

**Comment by nageshkumarmishra** [comment]
Please resolve the conflicts.

**Comment by ahiipsa** [comment]
> Please resolve the conflicts.

done

File: voice/voice-ai/x/OpenAIService/OpenAIStreamService.swift [hunk 1]
```diff
@@ -34,7 +34,11 @@ class OpenAIStreamService: NSObject, URLSessionDataDelegate {
         self.init(networkService: nil, completion: completion, temperature: temperature)
     }
 
-    init(networkService: NetworkService?, completion: @escaping (String?, Error?) -> Void, temperature: Double = 0.7) {
+    public func setNetworkService(urlSession: URLSession) {
+        self.networkService = urlSession
+    }
+
+    init(networkService: URLSession?, completion: @escaping (String?, Error?) -> Void, temperature: Double = 0.7) {
```
Comment (polymorpher) [inline voice/voice-ai/x/OpenAIService/OpenAIStreamService.swift:41]: Why are there two URLSession variables? One is already lazily initialized in class member

Comment (polymorpher) [reply voice/voice-ai/x/OpenAIService/OpenAIStreamService.swift:41]: I see, this one is only used in tests

**Comment by polymorpher** [comment]
Instead of adding initialization parameters just for tests, I wonder if we can set up a mock server instead - something that gives predictable, simple response with controlled latency. The tests would be more realistic as well

**Pull request merged**